testsuite/cluster-test/src/cluster_builder.rs

```diff
@@ -272,11 +323,174 @@ impl ClusterBuilder {
             })
         });
 
-        try_join!(
-            try_join_all(validators),
-            try_join_all(lsrs),
-            try_join_all(vaults),
-            try_join_all(fullnodes),
+        let validators = try_join_all(validators).await?;
+        let fullnodes = try_join_all(fullnodes).await?;
+        Ok((validators, lsrs, vaults, fullnodes))
+    }
+
+    async fn initialize_vault(&self, validator_index: u32, vault_node: &KubeNode) -> Result<()> {
+        let addr = vault_node.internal_ip.clone();
+        tokio::task::spawn_blocking(move || {
+            let mut vault_storage = Box::new(VaultStorage::new(
+                format!("http://{}:{}", addr, VAULT_PORT),
+                VAULT_TOKEN.to_string(),
+                None,
+                None,
+            ));
+            if validator_index == 0 {
+                vault_storage
+                    .create_key(format!("{}__{}", LIBRA_ROOT_NAME, LIBRA_ROOT_KEY).as_str())?;
+            }
+            vault_storage.create_key(
+                format!("{}__{}", validator_pod_name(validator_index), OWNER_KEY).as_str(),
+            )?;
```

Comment:
Do you mind adding `.map_err` with description of failed place in all places in this function where you use `?`

This seem to be useful pattern when debugging what is going wrong